core/src/main/kotlin/org/evomaster/core/problem/rest/service/SecurityRest.kt

```diff
@@ -604,6 +605,134 @@ class SecurityRest {
         }
     }
 
+
+    /**
+     * Check if there is a test case with a 403 and another one with a 200 without authentication.
+     * To check this, there must be a resource. It can either be newly created or already exist,
+     * such as during initialization. While the user who created this
+     * resource can access it (200), the other user cannot (403). However, if a 200 status code is
+     * returned when attempting to access the same resource without sending the authorization header,
+     * it indicates that authorization checks are wrongly ignored if no auth info is set.
+     * Example:
+     * POST /resources/ AUTH1 -> 201 (location header: /resources/42/)
+     * GET /resources/42/ AUTH1 -> 200
+     * GET /resources/42/ AUTH2 -> 403
+     * GET /resources/42/ NOAUTH -> 200
+     */
+    private fun handleForgottenAuthentication() {
+        actionDefinitions.forEach { op ->
+            val i403or401 = RestIndividualSelectorUtils.findIndividuals(
+                individualsInSolution,
+                op.verb,
+                op.path,
+                statusCodes = listOf(401,403)
+            )
+
+            if (i403or401.isEmpty()) {
+                return@forEach //there is not any protected resource for this path/verb.
+            }
+
+            val i2xx = RestIndividualSelectorUtils.findIndividuals(
+                individualsInSolution,
+                op.verb,
+                op.path,
+                statusGroup = StatusGroup.G_2xx,
+                authenticated = false
+            )
+
+            if(i2xx.isNotEmpty()){
+                // we have a 2xx without auth, so we can create a test case
+                // we can just take the smallest 403 or 401 and the smallest 2xx
+
+                // note: mocked external services might return 401/403 or 2xx without auth.
+                // in this case, we couldn't merge them because of "child already present" error.
+                // we need to fix this.
+
+                val first = i403or401.minBy { it.individual.size() }
+                val second = i2xx.minBy { it.individual.size() }.copy()
+
+                second.individual.removeHostnameResolutionAction(first.individual.seeAllActions().filter {
+                    it is HostnameResolutionAction
```

Comment:
i guess this is the problematic part of the code. i can check it carefully, but meanwhile you can fix the other comments in this review